task/sql-tasks.js

```diff
@@ -280,7 +501,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT DISTINCT A.CompanyName, A.PricePerItem, A.ProductName FROM( SELECT
+      Customers.CompanyName AS CompanyName,  CAST(OrderDetails.UnitPrice AS SIGNED) AS "PricePerItem", Products.ProductName
+       FROM Customers JOIN Orders ON Customers.CustomerID=Orders.CustomerID
+       JOIN OrderDetails ON Orders.OrderID=OrderDetails.OrderID
+       JOIN Products ON OrderDetails.ProductID=Products.ProductID ) As A 
+       JOIN(SELECT B.CompanyName, B.PricePerItem FROM(SELECT
+        Customers.CompanyName AS CompanyName,  MAX(CAST(OrderDetails.UnitPrice AS SIGNED)) AS "PricePerItem"
+       FROM Customers JOIN Orders ON Customers.CustomerID=Orders.CustomerID
+       JOIN OrderDetails ON Orders.OrderID=OrderDetails.OrderID
+       GROUP by Customers.CompanyName )AS B) B ON
+       A.CompanyName=B.CompanyName AND A.PricePerItem=B.PricePerItem 

```

Comment:
do not join on comanyname, it doesn't iniquely identify customer also it's not good for performance because this column odesn't have an index